Update the process for transitioning from Individual Contributor to Engineering Manager (#4227)
sourcegraph/about: handbook/engineering/engineering-management.md

- [The Five Dysfunctions of a Team](https://www.amazon.com/Five-Dysfunctions-Team-Leadership-Fable/dp/0787960756) - A model of how to think about what makes a highly effective team paired with a fictional story that illustrates the model.
- [The Leader’s Guide to Unconscious Bias](https://www.amazon.com/Leaders-Guide-Unconscious-Bias-High-Performing/dp/1982144319) ([facilitated small group discussions](../support/support-values-enablement.md#part-2-deep-dive-on-be-welcoming-and-inclusive).

## Promotion requests

To be documented.

## Compensation increase requests

To be documented.

> # Transitioning from Individual Contributor to Engineering Manager

Author: ```suggestion
## Transitioning from Individual Contributor to Engineering Manager
```

For teammates pursuing an Engineering Manager role but who lack the necessary people management experience, we have defined the following transitioning guide that affords the candidate a "manager in training" opportunity to try out being an Engineering Manager for a trial period. The trial period will allow the teammate to determine if they want to commit to it long-term and verify that they would be good in the role.

### Expressing interest in becoming an Engineering Manager

Teammates who have reached the [IC3 level](career-development/framework.md#ic3) and have a track record of good performance at Sourcegraph, having been employed for at least six months, are eligible for a manager in training trial.
The teammate should make their desire clear to their Engineering Manager so that when a relevant opportunity becomes available, they may be considered.

Opportunities for a manager in training trial can come about through a current Engineering Manager position becoming vacant, an existing team splitting into two, or an Engineering Manager going on long-term PTO. 

> When an opportunity arises, the teammates manager can nominate them for a manager in training trial.

Author: ```suggestion
When an opportunity arises, the teammate's manager can nominate them for a manager in training trial to the Director of Engineering for the relevant org.
```

There are some considerations to keep in mind to ensure that the teammate is set up for success:
- The teammate's existing team should be in a position for them not to be involved as an IC for the whole duration of the trial period. Managers in training should not fulfill a dual role.

> - The managed team should comply with our [minimum viable team](eng_org.md#minimum-viable-team) criteria.

Author: ```suggestion
- The team to be managed should comply with our [minimum viable team](eng_org.md#minimum-viable-team) criteria.
```

> - The managed team should have a dedicated and established Product Manager assigned to it.

Author: ```suggestion
- The team to be managed should have a dedicated and established Product Manager assigned to it.
```

> - The team to be managed should have a dedicated and established Product Manager assigned to it.

Author: ```suggestion
- The team to be managed should have a dedicated and established Product Manager assigned to it.
- The team to be managed should preferably not be one that the manager in training was an IC on.
```

### Manager in training trial period

During the trial period, the manager in training will assume full Engineering Manager responsibility as outlined in the [Engineering Manager responsibilities](roles.md#engineering-manager).

> The teammate's manager or another nominated manager will become their sponsoring manager for the duration of the trial period. The sponsoring manager must commit to providing extra support to the manager in training during the trial period, which will last 12 weeks. 

Reviewer: Does this have to be an EM or can it be a director too (e.g. in the case of the manager going on long-term PTO)?

Author: @felixfbecker to clarify are you asking whether a director could also be the sponsoring manager rather than just an EM? If so, then absolutely yes.

Reviewer: Yep, that's what I'm asking. If that's the case I'd suggest explicitly saying "manager or director" here to avoid misunderstandings.

Author: ```suggestion
The teammate's manager, director, or another nominated manager/director will become their sponsoring manager for the duration of the trial period. The sponsoring manager must commit to providing extra support to the manager in training during the trial period, which will last 12 weeks. 
```

The trial period may be extended if agreed to by the manager in training and the sponsoring manager. However, the trial period should not be extended to more than 16 weeks.

The sponsoring manager will give the manager in training active coaching and feedback during weekly 1-1s and conduct bi-weekly 1-1s with the manager in training's direct reports to collect feedback and set them up for success.
Halfway through their trial period, the manager in training will receive an interim performance evaluation to help them identify strengths and shortcomings as an engineering manager and allow them to adjust if needed.

The trial period can be canceled by the manager in training or the sponsoring manager at any time if it is not working out.

> ### Promotion to Engineering Manager

Reviewer: If the IC was level IC5 (or higher) before this wouldn't be a promotion but a lateral move. Should this perhaps not use the word "promotion"? Or would in that case the process below not be followed (no "promotion request" need to be filed)?

I think it's important to communicate that becoming an EM is first and foremost a role change and not _necessarily_ a promotion, since we do think of IC5 as equivalent in impact to an EM.

Author: I think it should still require a "promotion" request to be completed since it is a formal change. However, you call out an important point that the wording should reflect that it's a role change, not a promotion. I'll update the wording now.

- [The Five Dysfunctions of a Team](https://www.amazon.com/Five-Dysfunctions-Team-Leadership-Fable/dp/0787960756) - A model of how to think about what makes a highly effective team paired with a fictional story that illustrates the model.
- [The Leader’s Guide to Unconscious Bias](https://www.amazon.com/Leaders-Guide-Unconscious-Bias-High-Performing/dp/1982144319) ([facilitated small group discussions](../support/support-values-enablement.md#part-2-deep-dive-on-be-welcoming-and-inclusive).

## Promotion requests

To be documented.

## Compensation increase requests

To be documented.

## Transitioning from Individual Contributor to Engineering Manager

> For teammates pursuing an Engineering Manager role but who lack the necessary people management experience, we have defined the following transitioning guide that affords the candidate a "manager in training" opportunity to try out being an Engineering Manager for a trial period. The trial period will allow the teammate to determine if they want to commit to it long-term and verify that they would be good in the role.

Reviewer 2: Verbiage edit - feel free to take/leave what you'd like @jeanduplessis:

For teammates interested in becoming Engineering Managers who may not have prior experience in people management, we have defined a transition framework that affords this candidate a "Manager in training" opportunity for a trial period. This trial period will allow the teammate to gain exposure to management principles, determine if they would excel as an Engineering Manager, and help them envision committing to this career path long-term.

Author: Resolving feedback in https://github.com/sourcegraph/about/pull/4303/files

### Expressing interest in becoming an Engineering Manager

> Teammates who have reached the [IC3 level](career-development/framework.md#ic3) and have a track record of good performance at Sourcegraph, having been employed for at least six months, are eligible for a manager in training trial.

Reviewer 2: Verbiage edit - feel free to take/leave what you'd like @jeanduplessis:

**Prerequisites for a manager in training trial**: Teammates who have reached the [IC3 level](career-development/framework.md#ic3), have a track record of good performance at Sourcegraph, and have been teammates at Sourcegraph for at least six months.

The teammate should make their desire clear to their Engineering Manager so that when a relevant opportunity becomes available, they may be considered.

Opportunities for a manager in training trial can come about through a current Engineering Manager position becoming vacant, an existing team splitting into two, or an Engineering Manager going on long-term PTO. 

> When an opportunity arises, the teammate's manager can nominate them for a manager in training trial to the Director of Engineering for the relevant org.

Reviewer 2: Is a nomination as simple as Slacking the Director of Engineering in each respective org, or is there a more formal process to nominate someone?

There are some considerations to keep in mind to ensure that the teammate is set up for success:
- The teammate's existing team should be in a position for them not to be involved as an IC for the whole duration of the trial period. Managers in training should not fulfill a dual role.
- The team to be managed should comply with our [minimum viable team](eng_org.md#minimum-viable-team) criteria.
- The team to be managed should have a dedicated and established Product Manager assigned to it.
- The team to be managed should preferably not be one that the manager in training was an IC on.

### Manager in training trial period

During the trial period, the manager in training will assume full Engineering Manager responsibility as outlined in the [Engineering Manager responsibilities](roles.md#engineering-manager).

The teammate's manager, director, or another nominated manager/director will become their sponsoring manager for the duration of the trial period. The sponsoring manager must commit to providing extra support to the manager in training during the trial period, which will last 12 weeks. 

> The trial period may be extended if agreed to by the manager in training and the sponsoring manager. However, the trial period should not be extended to more than 16 weeks.

Reviewer 2: In what circumstances would this trial period be extended?

Author: Actioning feedback in https://github.com/sourcegraph/about/pull/4303/files

- [The Five Dysfunctions of a Team](https://www.amazon.com/Five-Dysfunctions-Team-Leadership-Fable/dp/0787960756) - A model of how to think about what makes a highly effective team paired with a fictional story that illustrates the model.
- [The Leader’s Guide to Unconscious Bias](https://www.amazon.com/Leaders-Guide-Unconscious-Bias-High-Performing/dp/1982144319) ([facilitated small group discussions](../support/support-values-enablement.md#part-2-deep-dive-on-be-welcoming-and-inclusive).

## Promotion requests

> To be documented.

Author: I will update this section in a follow-up PR

## Compensation increase requests

> To be documented.

Author: I will update this section in a follow-up PR

> ## Temporary management positions

Reviewer 3: Do we have a past example of a temporary management position? I am wondering if we can/should delete this bit and just focus the docs on how we train new managers.

Author: We have acting managers, which are temporary management positions all the time. However, I don't think it adds value in this context right now so will remove it.

We create temporary management positions when there is an organizational need. Some examples:

- A team has grown beyond the [maximum viable team guideline](eng_org.md#maximum-viable-team), and we decide to split the team.
- An Engineering Manager is out on extended PTO.
- We decide to form a new team from scratch. 

### Acting manager

An Acting Manager occupies a management role temporarily and will move back to their original position after a set amount of time or other conditions. An Acting Manager could be filling in for a vacant role while we hire or providing cover for another Engineering Manager on extended PTO.

### Manager in training

Manager in training positions are for teammates pursuing an Engineering Manager role long term but who lack the necessary people management experience. We want to allow teammates to consider if they would like to make the role switch and verify that they would be good in the role.

#### Requirements for a manager in training role

The following requirements are in place to ensure we set up the manager in training for success.

> 1. The candidate's manager or another nominated manager must commit to providing extra support to the manager in training for the duration of the trial period. They become the sponsoring manager.

Reviewer 3: > or another nominated manager

It feels a bit risky to me if the MIT is going to someone other than their sponsoring manager for advice. These conversations are important/useful for calibrating throughout the process

Author: The intention here was to say that the IC's line manager OR another manager (say a Director or EM in another org) can be the sponsoring manager. I'll rephrase it.

```suggestion
1. The candidate's manager or another nominated manager will become the sponsoring manager. The sponsoring manager must commit to providing extra support to the manager in training for the duration of the trial period.
```

1. The candidate should have a track record of good performance at Sourcegraph and should be employed at Sourcegraph for a minimum of 6 months before taking on the manager in training role.

> 1. The candidate's existing team must be in a position for them not to be involved for the whole duration of the trial period. Managers in training should not fulfill a dual role.

Reviewer 3: ```suggestion
1. The candidate's existing team must be in a position for them not to be involved as an IC for the whole duration of the trial period. Managers in training should not fulfill a dual role.
```

1. The managed team must comply with our [minimum viable team](eng_org.md#minimum-viable-team) criteria.
1. The managed team must have a dedicated and established Product Manager assigned to it.

#### Trial period

> 1. The trial period will last a minimum of 6 weeks and a maximum of 12 weeks.

Reviewer 3: When I wrote this in the JIRA ticket, my intention was to say "we should decide on a single fixed amount for all trials and it should be between 6-12 weeks". What do you think is the right duration for a trial?

Author: ```suggestion
1. The trial period will last 12 weeks but may be extended if agreed to by the manager in training and the sponsoring manager. The trial period should not be extended to more than 16 weeks.
```

I think 12 months is a good duration. Not too long, but long enough to get enough signal. 
There might be valid reasons to extend it, but we should cap it at 4 months.
